### PR TITLE
Pierre

### PR DESCRIPTION
**Pull request by Pierre-Malherbe**

### CODE REVIEW

File: src/main/java/com/ibreath/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@Api("Measure")
```
Comment (Pierre-Malherbe) [inline src/main/java/com/ibreath/controller/UserController.java]: Bad Api description

File: src/main/java/com/ibreath/annotation/ValidEmail.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+@Target({TYPE, FIELD, ANNOTATION_TYPE})
+@Retention(RUNTIME)
+@Constraint(validatedBy = EmailValidator.class)
+@Documented
+public @interface ValidEmail {
+    String message() default "Invalid email";
+    Class<?>[] groups() default {};
+    Class<? extends Payload>[] payload() default {};
+}
```
Comment (Pierre-Malherbe) [inline src/main/java/com/ibreath/annotation/ValidEmail.java]: Backspace

File: src/main/java/com/ibreath/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class UserController {
+
+    private UserService service;
+
+    @ApiOperation("Get measure Data")
```
Comment (Pierre-Malherbe) [inline src/main/java/com/ibreath/controller/UserController.java]: Bad Api description

File: pom.xml [hunk 4]
```diff
@@ -110,6 +110,23 @@
 			<version>1.10.19</version>
 			<scope>test</scope>
 		</dependency>
+
```
Comment (theobeuze) [inline pom.xml]: Why Blank line ?

File: src/main/java/com/ibreath/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @RequestMapping(value="/user/{userId}", method = RequestMethod.GET)
```
Comment (theobeuze) [inline src/main/java/com/ibreath/controller/UserController.java]: I think this path should be better /users/etc.. I just add the S to user. This change is also working for the other paths

File: src/main/java/com/ibreath/resource/model/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.ibreath.resource.model.repository;
+
+import com.ibreath.resource.model.entity.UserEntity;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository  extends CrudRepository<UserEntity, Long> {
+    UserEntity findById(Long id);
```
Comment (theobeuze) [inline src/main/java/com/ibreath/resource/model/repository/UserRepository.java]: This method is the same as findOne() so not necessary

File: src/main/java/com/ibreath/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            dto.setWeight(entity.getWeight());
+            dto.setHeight(entity.getHeight());
+            dto.setDecreaseRate(entity.getDecreaseRate());
+            dto.setGender(entity.getGender());
+            return dto;
+        } catch (Exception e) {
+            logger.fatal("Error with finding userId in getUser method : " + e.getMessage());
+            throw new Exception("The user doesen't exist");
+        }
+    }
```
Comment (theobeuze) [inline src/main/java/com/ibreath/service/UserService.java]: Why do you need a try catch ? you can just test your first result of the repository like 
if (entity == null) 
  Throw new etc...

**Pull request merged**